### PR TITLE
DD Forwarder expose ENV Vars, remove redundant lifecycles

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
* remove redundant lifecycle
* expose lambda env vars

## why
* env vars allow customization of DD lambda forwarder

## references
* closes #39 
* https://docs.datadoghq.com/logs/guide/forwarder/

### CODE REVIEW

**Review by milldr — APPROVED** [review]

**Comment by Benbentwo** [comment]
Adding Major label, this is not a major change but we are in the process of updating to 1.x.y releases across all modules

**Comment by Benbentwo** [comment]
Adding Major label, this is not a major change but we are in the process of updating to 1.x.y releases across all modules

**Comment by Benbentwo** [comment]
Adding Major label, this is not a major change but we are in the process of updating to 1.x.y releases across all modules

**Comment by Benbentwo** [comment]
Adding Major label, this is not a major change but we are in the process of updating to 1.x.y releases across all modules

**Comment by Benbentwo** [comment]
Adding Major label, this is not a major change but we are in the process of updating to 1.x.y releases across all modules

File: main.tf [hunk 1]
```diff
@@ -69,12 +69,20 @@ data "aws_iam_policy_document" "assume_role" {
 ######################################################################
 ## Create Lambda policy and attach it to the Lambda role
 
+resource "aws_iam_policy" "datadog_custom_policy" {
+  count  = local.lambda_enabled && length(var.lambda_policy_source_json) > 0 ? 1 : 0
+  name   = "DatadogForwarderCustomPolicy"
```
Comment (nitrocode) [inline main.tf:74]: @Benbentwo @milldr how come this name doesnt follow the null label?

**Comment by nitrocode** [comment]
@Benbentwo how come this was a major release?

**Pull request merged**